### ISSUE TITLE
Toothpick 2 to 3 migration issue: No factory could be found for class android.app.Activity

### ISSUE THREAD

**Issue by amaguran** [stephanenicolas/toothpick#394]
Hi,

First of all thank you for this great library. We've been using it since the first release (after migrating from Roboguice).

We wanted to migrate from Toothpick 2 to 3 and we encountered the "No factory could be found for class android.app.Activity" error. We are using the `SmoothieAndroidXActivityModule` class. The usage is simple. In our "base" activity (onCreate) we are calling:
```
Scope scope = Toothpick.openScopes(application, this);
scope.installModules(new SmoothieAndroidXActivityModule(this));
Toothpick.inject(activity, scope);
```

One of the activity's collaborators has a dependency to `android.app.Activity`.
```
toothpick.locators.NoFactoryFoundException: No factory could be found for class android.app.Activity. Check that the class has either a @Inject annotated constructor or contains @Inject annotated members.
        at toothpick.locators.FactoryLocator.getFactory(FactoryLocator.java:37)
        at toothpick.ScopeImpl.lookupProvider(ScopeImpl.java:359)
        at toothpick.ScopeImpl.getInstance(ScopeImpl.java:93)
        at toothpick.ScopeImpl.getInstance(ScopeImpl.java:84)
        at ActivityCollaborator__Factory.createInstance(ActivityCollaborator__Factory.java:12)
        at ActivityCollaborator__Factory.createInstance(ActivityCollaborator__Factory.java:8)
        at toothpick.InternalProvider.get(InternalProvider.java:174)
        at toothpick.InternalScopedProvider.get(InternalScopedProvider.java:80)
        at toothpick.ScopeImpl.getInstance(ScopeImpl.java:93)
        at toothpick.ScopeImpl.getInstance(ScopeImpl.java:84)
...
```

I don't understand why the binding from SmoothieAndroidXActivityModule (bind(Activity.class).toInstance(activity)) does not work.

I could not find any indication in the current issues, so please provide some hints in which direction I should look into.

Thank you.
Alex

**Comment by stephanenicolas** [comment]
It should definitely work. Are you sure you use the activity scope you created to inject your collaborator ? How is it injected from within the activity?

**Comment by amaguran** [comment]
After some debugging I've noticed that the Activity instance lookup reaches the parent scope (app scope) instead of the activity scope (if I invoke scope.getInstance(Activity.class), in the above code, it returns the proper instance of the activity). This seems to happen because the ActivityCollaborator's provider is bound at the app level module which is installed in the app scope. We did some modularization and some classes have remained at the app level. However, is this the intended behavior of TP3?
